src/sql/orderbook/prod_order_rewards.sql

```diff
@@ -33,293 +33,148 @@ with trade_hashes as (
-                    -- at most charge a fraction of volume
-                    fp.surplus_factor / (1 - fp.surplus_factor) * surplus -- charge a fraction of surplus
-                )
-            END
-            WHEN fp.kind = 'priceimprovement' THEN CASE
-                WHEN os.kind = 'sell' THEN
-                LEAST(
-                    -- at most charge a fraction of volume
-                    fp.price_improvement_max_volume_factor / (1 - fp.price_improvement_max_volume_factor) * os.buy_amount,
-                    -- charge a fraction of price improvement, at most 0
-                    GREATEST(
-                        fp.price_improvement_factor / (1 - fp.price_improvement_factor) * price_improvement
-                        ,
-                        0
-                    )
-                )
-                WHEN os.kind = 'buy' THEN LEAST(
-                    -- at most charge a fraction of volume
-                    fp.price_improvement_max_volume_factor / (1 + fp.price_improvement_max_volume_factor) * os.sell_amount,
-                    -- charge a fraction of price improvement
-                    GREATEST(
-                        fp.price_improvement_factor / (1 - fp.price_improvement_factor) * price_improvement,
-                        0
-                    )
-                )
-            END
-            WHEN fp.kind = 'volume' THEN CASE
-                WHEN os.kind = 'sell' THEN
-                    fp.volume_factor / (1 - fp.volume_factor) * os.buy_amount
-                WHEN os.kind = 'buy' THEN
-                    fp.volume_factor / (1 + fp.volume_factor) * os.sell_amount
-            END
-        END AS protocol_fee_first,
-        os.surplus_token AS protocol_fee_token
+            WHEN tdu.partner_fee_recipient IS NOT NULL THEN tdu.second_protocol_fee_amount
+            ELSE 0
+        END AS partner_fee,
+        tdu.surplus_token AS protocol_fee_token,
+        pfk.kind as protocol_fee_kind
     FROM
-        order_surplus os
-        JOIN fee_policies_first fp -- contains protocol fee policy
-        ON os.auction_id = fp.auction_id
-        AND os.order_uid = fp.order_uid
-),
-order_surplus_intermediate as (
-    select
-        auction_id,
-        order_uid,
-        CASE
-            WHEN kind = 'sell' then sell_amount
-            ELSE sell_amount - protocol_fee_first
-        END as sell_amount,
-        CASE
-            WHEN kind = 'sell' then buy_amount + protocol_fee_first
-            ELSE buy_amount
-        END as buy_amount,
-        surplus + protocol_fee_first as surplus,
-        price_improvement + protocol_fee_first as price_improvement,
-        protocol_fee_kind_first,
-        protocol_fee_first,
-        partner_fee_recipient
-    from order_protocol_fee_first
+        trade_data_unprocessed tdu LEFT OUTER JOIN protocol_fee_kind pfk ON tdu.order_uid = pfk.order_uid AND tdu.auction_id = pfk.auction_id 
```

Comment:
To make it consistent with the other formating:
```suggestion
        trade_data_unprocessed tdu
            LEFT OUTER JOIN protocol_fee_kind pfk
            ON tdu.order_uid = pfk.order_uid AND tdu.auction_id = pfk.auction_id 
```